### PR TITLE
mech nitrogen filtering 2

### LINKED ISSUE #19863: Vim/Hamtr suffucate user
## Description
Mouse/Hammy in Vim/Hamtr cannot breathe after a while.

**Reproduction**
Get in Vim/Hamtr as lil critter, wait a while, gasping begins.

**Screenshots**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
better version of last one
if there is less than 21% oxygen it will start filtering out nitrogen
fixes #19863

## Why / Balance
bug making vim eventual deathtrap

## Technical details
added `TargetOxygen` which is checked when filtering against the current oxygen ratio. if there is not enough oxygen it will filter out `OverflowGases`, which includes nitrogen for basemech.
also moved everything to atmos so it can be reused for other things like gas masks
uses `GetFilterAirEvent` so it doesnt depend on mech

## Media
trust
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
`MechAirFilterComponent` renamed to `AirFilterComponent`
`MechAirIntakeComponent` renamed to `AirIntakeComponent`
`AirFilterComponent` now has the required field `overflowGases` which should be the same as `gases` plus a gas to filter out to maintain oxygen ratio. Adding nitrogen is normal for this.

**Changelog**
no cl no fun

### CODE REVIEW

**Comment by Partmedia** [comment]
Perhaps at this point it's better to start seeing where code can be shared with vents/scrubbers instead of doing a wholesale copy/pasta.

**Comment by deltanedas** [comment]
with these changes it would also be an air alarm and air alarm would need to support saving gases per warning/danger

File: Resources/Prototypes/Entities/Objects/Specific/Mech/mechs.yml [hunk 1]
```diff
@@ -17,6 +17,17 @@
...
+    overflowGases:
+    - 1
+    - 2
+    - 3
+    - 4
+    - 5
+    - 6
+    - 7
+    - 8
+    #- 9 TODO: fusion
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Specific/Mech/mechs.yml:30]: Can't you just use the enum name here.

Comment (deltanedas) [reply Resources/Prototypes/Entities/Objects/Specific/Mech/mechs.yml:30]: other atmos prototypes like reactions dont so i assume not

**Comment by deltanedas** [comment]
@metalgearsloth merg

**Comment by Emisse** [comment]
Thank you for making it generic

**Comment by deltanedas** [comment]
i was gonna make followup gas mask filtering pr but the body code trolled me or something :trollface:

**Pull request merged**